resources/assets/js/components/youtube-embed/YoutubeEmbed.vue

```diff
@@ -0,0 +1,117 @@
+            background-repeat: no-repeat;
+            height: 72px;
+            width: 72px;
+            left: 50%;
+            top: 50%;
+            margin-left: -36px;
+            margin-top: -36px;
+            position: absolute;
+            cursor: pointer;
+        }
+    }
+</style>
+
+<template>
+    <div class="youtube-player">
+        <template v-if="queuedToPlay">
+            <iframe :id="playerId" :src="videoSource" ref="ytiframe" frameborder="0" allow="autoplay; encrypted-media"
+                    allowfullscreen></iframe>
+        </template>
+        <template v-else>
+            <img :src="thumbSource">
+            <div class="yt-play-button" @click="play"></div>
+        </template>
+    </div>
+</template>
+
+<script>
+    export default {
+        props: {
+            videoId: {
+                type: String,
+                required: true
+            }
+        },
+        data() {
+            return {
+                queuedToPlay: false,
+                player: null,
+                playerId: `ytplayer-${this.videoId}`,
+                videoSource: `https://www.youtube.com/embed/${this.videoId}?autoplay=1&amp;enablejsapi=1`,
+                thumbSource: `https://i.ytimg.com/vi/${this.videoId}/hqdefault.jpg`
+            }
+        },
+        methods: {
+            play: function() {
+                this.queuedToPlay = true;
+
+                this.getIframeApi();
+
+                let makeYtPlayerInterval = setInterval(() => {
+                    if (typeof YT !== 'undefined') {
+                        this.player = new YT.Player(this.playerId);
+                        // Without setTimeout, playVideo() is not a function, but why?
+                        setTimeout(() => {
+                            this.player.playVideo();
+                        }, 1000);
+                        clearInterval(makeYtPlayerInterval);
+                    }
+                }, 100)
+            },
+            getIframeApi: function() {
+                // We need to change template to show iframe.
+                // Tried to use axios.get but get:
+                // Failed to load https://www.youtube.com/iframe_api: Response to preflight request doesn't pass access control
+                // check: No 'Access-Control-Allow-Origin' header is present on the requested resource.
+                // Origin 'https://dev.gzero.pl' is therefore not allowed access. The response had HTTP status code 405.
```

Comment:
We want API to be inserted as a tag to be available for the other components which may be needed it.

If we use e.g 3 components, the first one should create a tag and other ones should just use the API, without creating it again.